### PR TITLE
[botan] update to version 2.17.3

### PR DESCRIPTION
**Pull request by agrandville**
feat add debug symbols

- #### triplets supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  YES

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  YES

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by autoantwort** [comment]
Please run `./vcpkg format-manifest --all` :)

File: ports/botan/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "botan",
+  "version-string": "2.17.3",
```
Comment (NancyLi1013) [inline ports/botan/vcpkg.json:3]: It would be better to served by [version-semver](https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.reference.md#version-semver).

File: versions/b-/botan.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "bc62b88dc0ea58b6121cd0f1d628d1384869af45",
+      "version-string": "2.17.3",
```
Comment (NancyLi1013) [inline versions/b-/botan.json:5]: The same as above.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]
Could you please update `botan `as `${PORT}` in the following codes?

```
vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/botan)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/botan-${BOTAN_VERSION}/ ${CURRENT_PACKAGES_DIR}/share/botan/)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/botan/license.txt ${CURRENT_PACKAGES_DIR}/share/botan/copyright)
```

**Comment by NancyLi1013** [comment]
Please run` ./vcpkg format-manifest ./ports/botan/vcpkg.json
` to format vcpkg.json file.

File: ports/botan/portfile.cmake [hunk 3]
```diff
@@ -77,7 +76,8 @@ function(BOTAN_BUILD BOTAN_BUILD_TYPE)
                             "--distribution-info=vcpkg ${TARGET_TRIPLET}"
                             --prefix=${BOTAN_FLAG_PREFIX}
                             --with-pkg-config
-                            --link-method=copy)
+                            --link-method=copy
+                            --with-debug-info)
```
[the remark this answers is not in the document]
Comment (agrandville) [reply ports/botan/portfile.cmake:80]: I activate this flag to generate PDB file including for the release build

File: versions/b-/botan.json (hunk 2 above)
Comment (NancyLi1013) [inline versions/b-/botan.json:5]:
```suggestion
      "version": "2.17.3",
```
This also should be `version`.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your update @agrandville.

For the failures on osx pipeline, there is something wrong with our CI, which is not related with this PR. I will rerun it once solved.

**Comment by NancyLi1013** [comment]
The failures caused by `popsift `will be fixed by #17277.

**Comment by NancyLi1013** [comment]
```
Error: While reading versions for port botan from file: C:\a\1\s\versions\b-\botan.json
       File declares version `2.17.3` with SHA: f355b4fc82c9cb1c7de67be66ac517931a488f01
       But local port with the same verion has a different SHA: 2d15c43909a9409da601e7b01a9a3a288c50bfb9
       Please update the port's version fields and then run:

           vcpkg x-add-version botan

       to add a new version.
```
Please run `vcpkg x-add-version botan` again to update versions.

Note: Have you tested the usage of `botan`? I noticed that added botan-config.cmake file in the last commit.

**Comment by agrandville** [comment]
I'm upgrading the cmake scripts of opendnssec/SoftHSMv2 project who use vcpkg botan.  
For now, SoftHSM compile, with botan crypto backend and vcpkg toolchain, fails due  to missing BotanConfig.cmake file.

File: versions/b-/botan.json [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "f355b4fc82c9cb1c7de67be66ac517931a488f01",
+      "version": "2.17.3",
+      "port-version": 1
+    },
+    {
+      "git-tree": "2d15c43909a9409da601e7b01a9a3a288c50bfb9",
+      "version": "2.17.3",
+      "port-version": 0
+    },
```
Comment (NancyLi1013) [inline versions/b-/botan.json:12]:
```suggestion
```
This should not be existed.

File: versions/b-/botan.json [hunk 5]
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "f355b4fc82c9cb1c7de67be66ac517931a488f01",
+      "version": "2.17.3",
+      "port-version": 1
```
Comment (NancyLi1013) [inline versions/b-/botan.json:6]:
```suggestion
      "port-version": 0
```

File: ports/botan/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "botan",
+  "version": "2.17.3",
+  "port-version": 1,
```
Comment (NancyLi1013) [inline ports/botan/vcpkg.json:4]:
```suggestion
  "port-version": 0,
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
You can use `vcpkg-x-add-version --overwrite-version botan` to update the version files.

File: versions/baseline.json [hunk 7]
```diff
@@ -1005,8 +1005,8 @@
       "port-version": 2
     },
     "botan": {
-      "baseline": "2.16.0",
-      "port-version": 0
+      "baseline": "2.17.3",
+      "port-version": 1
```
Comment (NancyLi1013) [inline versions/baseline.json:1009]:
```suggestion
      "port-version": 0
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
@agrandville could you please merge with latest?

**Review by NancyLi1013 — APPROVED** [review]

File: ports/botan/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,51 @@
+project(botan)
```
Comment (strega-nil-ms) [inline ports/botan/CMakeLists.txt:1]: Why is this file here? it doesn't seem to be used by `portfile.cmake`...

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]

File: ports/botan/CMakeLists.txt (hunk 8 above)
Comment (NancyLi1013) [reply ports/botan/CMakeLists.txt:1]: Thanks for pointing this out @strega-nil-ms. Sorry, I missed this point before.

**Comment by NancyLi1013** [comment]
@agrandville 

Could you please address the review suggestions? Thanks.

**Comment by PhoebeHui** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.
